crates/ruma-api-macros/src/api/metadata.rs

```diff
@@ -125,32 +170,43 @@ enum Field {
     Path,
     RateLimited,
     Authentication,
+    Added,
+    Deprecated,
+    Removed,
+}
+
+macro_rules! map_kw {
+    ($input:ident, $kw:path => $el:expr, $($next_kw:path => $next_el:expr,)*) => {
+        let lookahead = $input.lookahead1();
+
+        if lookahead.peek($kw) {
+            let _: $kw = $input.parse()?;
+            Ok($el)
+        }
+        $(
+            else if lookahead.peek($next_kw) {
+                let _: $next_kw = $input.parse()?;
+                Ok($next_el)
+            }
+        )*
+        else {
+            Err(lookahead.error())
+        }
+    };
 }
 
 impl Parse for Field {
     fn parse(input: ParseStream<'_>) -> syn::Result<Self> {
-        let lookahead = input.lookahead1();
-
-        if lookahead.peek(kw::description) {
-            let _: kw::description = input.parse()?;
-            Ok(Self::Description)
-        } else if lookahead.peek(kw::method) {
-            let _: kw::method = input.parse()?;
-            Ok(Self::Method)
-        } else if lookahead.peek(kw::name) {
-            let _: kw::name = input.parse()?;
-            Ok(Self::Name)
-        } else if lookahead.peek(kw::path) {
-            let _: kw::path = input.parse()?;
-            Ok(Self::Path)
-        } else if lookahead.peek(kw::rate_limited) {
-            let _: kw::rate_limited = input.parse()?;
-            Ok(Self::RateLimited)
-        } else if lookahead.peek(kw::authentication) {
-            let _: kw::authentication = input.parse()?;
-            Ok(Self::Authentication)
-        } else {
-            Err(lookahead.error())
+        map_kw! { input,
```

Comment:
I don't think the macro definition plus its single use is actually (much) shorter than writing it out. It may be easier to read, but only once you've really understood what the macro does. I would prefer not having it.